### PR TITLE
Make torch.library decorators return function

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79051
* #79042
* #79026
* #78997
* __->__ #78996
* #78993
* #78992

This allows for chaining.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/library.py [hunk 1]
```diff
@@ -91,10 +91,12 @@ def __del__(self):
 def impl(lib, name, dispatch_key=""):
     def wrap(f):
```
Comment (justinchuby) [inline torch/library.py:92]: Would it make sense to use `@functools.wraps` to preserve the function info?

Comment (ezyang) [reply torch/library.py:92]: ye

Comment (justinchuby) [reply torch/library.py:92]: Ah that’s right

**Review by Chillee — APPROVED** [review]
LGTM!

**Pull request closed without merging**